components/listings-with-map.tsx

```
@@ -119,6 +132,110 @@ export default function ListingsWithMap() {
+    if (criteria.petFriendly !== undefined) {
+      filtered = filtered.filter(listing =>
+        listing.petFriendly === criteria.petFriendly
+      )
+    }
+
+    // Add distance calculation if radius is specified
+    if (criteria.location.coordinates && criteria.location.radius) {
+      filtered = filtered.map(listing => ({
+        ...listing,
+        distance: calculateDistance(
+          criteria.location.coordinates!.lat,
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)